### PR TITLE
Drivers: led: lp5562: re-use already linked engine

### PR DESCRIPTION
**Pull request by cx-anuj-pathak**
Right now blink api find and assign a engine that is not executing 
This is an issue, if 3 led are in blink already. 
That is new blink api call, can not find any idle engine (all are in executing state) and return with error. 
However we can simply reuse engine, already linked to the led

### CODE REVIEW

**Review by ubieda — DISMISSED** [review]

**Comment by simonguinot** [comment]
@RickBruyninckx @JoeHut please can you have a look at this ?

File: drivers/led/lp5562.c [hunk 1]
```diff
@@ -769,15 +769,24 @@ static int lp5562_led_blink(const struct device *dev, uint32_t led,
...
+	ret = lp5562_get_led_source(dev, led, &engine);
 	if (ret) {
 		return ret;
 	}
 
-	ret = lp5562_set_led_source(dev, led, engine);
-	if (ret) {
-		LOG_ERR("Failed to set LED source.");
-		return ret;
+	/* find and assign new engine only if not using any */
```
Comment (simonguinot) [inline drivers/led/lp5562.c:778]: Please, fix this comment too.

**Comment by RickBruyninckx** [comment]
> @RickBruyninckx @JoeHut please can you have a look at this ?

I currently don't have the hardware with me to test it but the code looks fine.

**Review by simonguinot — COMMENTED** [review]
Hi @cx-anuj-pathak,

Thanks for this patch and for taking care of this driver. I don't know this controller but it looks good to me.

See below for my comments. Additionally can you rephrase the commit message ? Some parts such as "that is new blink api so not find any idle engine" are not clear. Also capital letters are missing at the beginning of sentences, period is missing after "not executing", and API should be written in capital letters (acronym).

**Review by ubieda — APPROVED** [review]

**Review by simonguinot — APPROVED** [review]
Thanks !
